packages/angular-table/src/index.ts

```diff
@@ -1,96 +1,78 @@
-        () => {
-          return {
-            state: {},
-            onStateChange: () => {},
-            renderFallbackValue: null,
-            ...options(),
-          }
-        }
-      )
+  return lazyInit(() => {
+    // Compose table resolved options as computed.
+    // This will allow the effect to be triggered when options are updated.
+    const resolvedOptionsSignal = computed<TableOptionsResolved<TData>>(() => ({
+      state: {},
+      onStateChange: () => {},
+      renderFallbackValue: null,
+      ...options(),
+    }))
 
-      const notifier = signal([], { equal: () => false })
-      const table = createTable(untracked(resolvedOptionsSignal))
-      const state = signal(table.initialState)
+    const table = createTable(resolvedOptionsSignal())
 
-      function updateOptions() {
-        const tableState = untracked(state)
-        const resolvedOptions = untracked(resolvedOptionsSignal)
-        untracked(() => {
-          table.setOptions(prev => ({
-            ...prev,
-            ...resolvedOptions,
-            state: { ...tableState, ...resolvedOptions.state },
-            onStateChange: updater => {
-              const value =
-                updater instanceof Function ? updater(tableState) : updater
-              state.set(value)
-              resolvedOptions.onStateChange?.(updater)
-            },
-          }))
-        })
-      }
+    // By default, manage table state here using the table's initial state
+    const state = signal<TableState>(table.initialState)
 
-      updateOptions()
+    function updateOptions() {
+      const tableState = state()
+      const resolvedOptions = resolvedOptionsSignal()
+      table.setOptions(prev => ({
+        ...prev,
+        ...resolvedOptions,
+        state: { ...tableState, ...resolvedOptions.state },
+        onStateChange: updater => {
+          const value =
+            updater instanceof Function ? updater(tableState) : updater
+          state.set(value)
+          resolvedOptions.onStateChange?.(updater)
+        },
+      }))
+    }
 
-      let firstRender = true
-      effect(() => {
-        void [state(), resolvedOptionsSignal()]
-        if (firstRender) {
-          return (firstRender = false)
-        }
-        untracked(() => {
-          updateOptions()
-          notifier.set([])
-        })
-      })
+    // set table options again when options are updated
+    effect(() => updateOptions(), { injector })
 
-      const tableSignal = computed(
-        () => {
-          notifier()
-          return table
-        },
-        {
-          equal: () => false,
-        }
-      )
+    // set table options for the first time
+    updateOptions()
+
+    // convert table instance to signal for proxify to listen to any table state and options changes
+    const tableSignal = computed(
+      () => {
+        void [state(), resolvedOptionsSignal()]
+        return table
```

Comment:
I just have a doubt, considering that we have an effect which listen on both state and resolvedOptions and then we call `updateOptions`, does this `computed` get called **after** the table has been updated? The `notifier` variable was present in order to trigger this signal manually after all operations has been done

Reply:
`effect` immediately executes while `computed` lazily executed